File: problems/prob6 - Sum square difference/sln.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(square_of_sum(100) - sum_of_square(100))
}

func square_of_sum(n int) int {
	sum := (n / 2) * (1 + n);
	return sum * sum;
}

func sum_of_square(n int) int {
	return (n * (n + 1) * (2 * n + 1)) / 6;
}

File: problems/prob6 - Sum square difference/sln.java
import java.lang.Math;

class Main {
    public static void main(String[] args) {
        System.out.println(square_of_sum(100) - sum_of_square(100));
    }

    public static Integer square_of_sum(int n) {
        Integer sum = (n / 2) * (1 + n);
        return sum * sum;
    }

    public static Integer sum_of_square(int n) {
        return (n * (n + 1) * (2 * n + 1)) / 6;
    }
}
